pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentpruner/SegmentPrunerFactory.java

```diff
@@ -45,30 +50,41 @@ private SegmentPrunerFactory() {
 
   public static List<SegmentPruner> getSegmentPruners(TableConfig tableConfig,
       ZkHelixPropertyStore<ZNRecord> propertyStore) {
-    RoutingConfig routingConfig = tableConfig.getRoutingConfig();
     List<SegmentPruner> segmentPruners = new ArrayList<>();
-    // Always prune out empty segments first
-    segmentPruners.add(new EmptySegmentPruner(tableConfig, propertyStore));
+    boolean isKinesisEnabled = isKinesisEnabled(tableConfig);
```

Comment:
this is leaking immensely specific information into the broker about Kinesis and the behavior of empty segments. Plus, adding kinesis plugin dependency in pinot-broker is not the best.. 
How about one of these options:
1. Add validations to TableConfigUtils.validate, to check that a kinesis stream table has this pruner added (or if there's any logic in that path which decorates the table config)
2. Move this mthod `isKinesisEnabled` to `TableConfigUtils` ad rename it as `needsEmptySegmentPruner`. Part of that, check if routingTypes already has EmptySegmentPruner, if not check if kinesis. Possibly even add "needsEmptySegmetPruner" to StreamConfig

Reply:
@npawar 
There are two TableConfigUtils, 
* org.apache.pinot.segment.local.utils.TableConfigUtils;
* org.apache.pinot.common.utils.config.TableConfigUtils;

Which one is preferred to put `needsEmptySegmentPruner `?

Reply:
cc @snleee There are two TableConfigUtils,
```
org.apache.pinot.segment.local.utils.TableConfigUtils;
org.apache.pinot.common.utils.config.TableConfigUtils;
```
* If we put the `needsEmptySegmentPruner ` function in `org.apache.pinot.segment.local.utils.TableConfigUtils;`, then we need add  `pinot-kinesis` dependency to package `pinot-segment-local`
* If we put the `needsEmptySegmentPruner ` function in `org.apache.pinot.common.utils.config.TableConfigUtils`, then we need add  `pinot-kinesis` dependency to package `pinot-common`

kinesis plugin dependency will not been added to pinot-broker, but must be add to either `pinot-segment-local` or `pinot-common`

Reply:
add it to the one that has all the table config validations

Reply:
But neither pinot-common nor pinot-segment-local should depend on plugin. Do this instead `add "needsEmptySegmetPruner" to StreamConfig`